deps/fast_float_c_interface/fast_float_strtod.cpp

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Valkey Contributors.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD 3-Clause
+ */
+
+#include "../fast_float/fast_float.h"
+#include <cerrno>
+
+extern "C"
+{
+    double fast_float_strtod(const char *str, const char** endptr)
+    {
+        double temp = 0;
+        auto answer = fast_float::from_chars(str, str + strlen(str), temp);
+        if (answer.ec != std::errc()) {
+            errno = (answer.ec == std::errc::result_out_of_range) ? ERANGE : EINVAL;
+        } else {
+            errno = 0;
```

Comment:
Library functions (like stdlib's strtod and others) don't set errno to zero on success. That's something the caller can do, and we do it in the calling code.

```suggestion
```